backend/src/main/java/org/cryptomator/hub/api/VaultResource.java

```diff
@@ -247,10 +248,15 @@ public List<DeviceResource.DeviceDto> getDevicesRequiringAccessGrant(@PathParam(
 	@Operation(summary = "get the device-specific masterkey")
 	@APIResponse(responseCode = "200")
 	@APIResponse(responseCode = "402", description = "number of effective vault users exceeds available license seats")
-	@APIResponse(responseCode = "403", description = "device not authorized to access this vault")
-	@APIResponse(responseCode = "404", description = "unknown device")
+	@APIResponse(responseCode = "403", description = "vault is archived or device not authorized to access this vault")
+	@APIResponse(responseCode = "404", description = "vault or device not found")
```

Comment:
while physically still present, I would argue that after archiving, the vault does no longer "exist" from an "unlock perspective", hence I would rather throw a NotFoundException.

Reply:
Returning 404 only for the GET request would make our API inconsistent: For an archived vault v1 and device d1 `GET /v1/keys/d1` returns 404, but at the same time, the API would return 409 (Conflict) on `PUT /v1/keys/d1`.

I'd would suggest to either keep everything as it is or also adjust the PUT endpoint.

Reply:
After a discussion we setteled to the following:
1. Symmetry in the API is nice, but it is our backend, so this is not necessary, but...
2. there could be a race condition: You already unlocked the vault in the desktop app, but someone else archived it. Still you are going to register your device
3. PUT and GET for `/vaultId/keys/deviceId` will return 410 if the vault is archived

Reply:
Fixed in https://github.com/cryptomator/hub/pull/202/commits/c885038d5be367e2243c417cbcbebe5772850757